experimental/veggieseasons/lib/main.dart

```diff
@@ -80,6 +82,7 @@ class _RestorableAppState extends RestorableListenable<AppState> {
     final favorites = (data as List<dynamic>).cast<int>();
     for (var id in favorites) {
       appState.setFavorite(id, true);
+      preferences.setFavioriteVeggie(id, true);
```

Comment:
The approach that you've taken looks designed to minimize the changes to the existing codebase, which is an understandable goal. I think, though, that it's resulted in a solution that's spreading information around more than it needs to be.

If you look at the AppState class, you'll notice that it doesn't store any of the information currently kept in preferences (desired calories and preferred categories). Likewise, the Preferences class didn't store any information about the list of veggies.

In this change, however, information about favorite veggies is getting spread among both classes, which is resulting in strange situations like this one (setFavorite and setFavoriteVeggie both need to be called, even though the concern the same information).  

A better solution would be to find a way to remove the favorites information from AppState entirely and keep it solely in Preferences, or devise some other way the maintain separation of concern.

This would likely result in a much more complicated PR than you had originally anticipated, so it's perfectly reasonable if you'd prefer to just close this.

Reply:
Actually, As you said I tried to solve the issue with minimum changes in the existing code since I haven't discussed the right approach to solve it. 

I would love to learn and contribute to solving this issue. That's why I have set it to WIP and working on separating favorites information from the AppState.